### ISSUE TITLE
Edit X-Label and reindex notmuch database simultaneously

### ISSUE THREAD

**Issue by asakura42** [neomutt/neomutt#3024]
Hello. I want to create macro to do 2 things at same time: 1) <edit-label> to edit x-label and 2) `notmuch reindex` that message or thread to provide new info to notmuch for proper working with virtual mailboxes and search. Please, help me combine these two commands, I don't know how to get selected message's id/thread.

P.S. Yep, I know that I can edit notmuch tags, but it's not very convenient because it works only in virtual mailboxes

Maybe some other option exist to reload notmuch databases because they breaks after adding x-label and message became deleted from notmuch-backed virtual mailbox.

P.P.S. Most interesting thing is that message got deleted from virtual mailbox even after reindexing if I delete X-Label from message.

**Comment by Austin-Ray** [comment]
I don't think this is possible with macros as upstream Mutt's macro documentation[0] states:

> Basically you can do everything you would enter interactively in a macro, too. With one exception: You can't prompt the user for input.

And, to my knowledge, NeoMutt's macro code has the same exception.

@flatcap: Can you confirm if we have that exception?

[0] https://gitlab.com/muttmua/mutt/-/wikis/MuttGuide/Macros

**Comment by flatcap** [comment]
> @flatcap: Can you confirm if we have that exception?

Yes, you can't prompt the user.
Adding a `<ask-the-user>` function has been suggested, but nobody's stepped forward to implement it.

However...

You can mess around with the `source` command.
```
source "myscript.sh|"
```
means run the script, then process its output as NeoMutt config.

**myscript.sh**
```
#!/bin/bash
read -p "Enter a message: " MSG
echo "echo \"$MSG\""
```

In NeoMutt:
```
:source myscript.sh|
```

In terminal:
```
Enter a message: hello world!
```

In NeoMutt message window:
```
hello world!
```
